excel/src/main/java/org/apache/metamodel/excel/ExcelUpdateCallback.java

```diff
@@ -78,7 +87,24 @@ public ExcelDataContext getDataContext() {
 
     protected void close() {
         if (_workbook != null) {
-            ExcelUtils.writeWorkbook(_dataContext, _workbook);
+
+            final Resource resource;
+            if(_dataContext.getResource() instanceof FileResource){
```

Comment:
I think it was an attempt to avoid file locking issues I saw with File-based Excel writing on close. In the end of my attempts there was quite a bit of throwing stuff at the wall to see what stuck (I couldn't test locally, as only Travis would fail).